compute/xmpp_wikibot/wikibot.py

```diff
@@ -94,22 +97,27 @@ def message(self, msg):
         """
         if msg['type'] in ('chat', 'normal'):
             msg_body = msg['body']
-            logging.info('Message sent was: {}'.format(msg_body))
             encoded_body = urllib.quote_plus(msg_body)
             svrResponse = requests.get(
                 'https://en.wikipedia.org/w/api.php?'
-                'action=parse&prop=sections&format=json&page={}'.format(
-                    encoded_body))
+                'action=query&list=search&format=json&srprop=snippet&'
+                'srsearch={}'.format(encoded_body))
             doc = json.loads(svrResponse.content)
-            try:
-                page_id = str(doc['parse']['pageid'])
-                defn_url = 'https://en.wikipedia.org/?curid={}'.format(page_id)
-                msg.reply('find out more about: "{}" here: {}'.format(
-                    msg_body, defn_url)).send()
-            except KeyError as e:
-                logging.info('key error: {0}'.format(e))
+
+            results = doc.get('query', {}).get('search')
+            if not results:
                 msg.reply('I wasn\'t able to locate info on "{}" Sorry'.format(
                     msg_body)).send()
+                return
+
+            snippet = unicode(results[0]['snippet'])
```

Comment:
Hrm... I don't even remember why I was casting it to unicode anyway (especially since the encoding in wikipedia is fickle, I think). I'm just going to remove the unicode cast.